### PR TITLE
datapath: allow local NodePort traffic for `eni+` container interfaces with CNI chaining

### PR DESCRIPTION
**Pull request by ti-mo**
AWS CNI chaining yields container interface names like 'eni621c0fc8425', not the usual 'lxcXYZ'. This causes packets for local endpoints to be dropped in CILIUM_FORWARD when they are called through a NodePort.

Before the patch, the CILIUM_FORWARD chain looks like this:

```
-A CILIUM_FORWARD -o cilium_host -m comment --comment "cilium: any->cluster on cilium_host forward accept" -j ACCEPT
-A CILIUM_FORWARD -i cilium_host -m comment --comment "cilium: cluster->any on cilium_host forward accept (nodeport)" -j ACCEPT
-A CILIUM_FORWARD -i lxc+ -m comment --comment "cilium: cluster->any on lxc+ forward accept" -j ACCEPT
-A CILIUM_FORWARD -i cilium_net -m comment --comment "cilium: cluster->any on cilium_net forward accept (nodeport)" -j ACCEPT
-A CILIUM_FORWARD -o lxc+ -m comment --comment "cilium: any->cluster on lxc+ forward accept" -j ACCEPT
-A CILIUM_FORWARD -i lxc+ -m comment --comment "cilium: cluster->any on lxc+ forward accept (nodeport)" -j ACCEPT
```

This doesn't match any packets to or from `eni+` container interfaces, letting them fall through to the `KUBE-FORWARD` chain instead:

```
-A FORWARD -m comment --comment "cilium-feeder: CILIUM_FORWARD" -j CILIUM_FORWARD
-A FORWARD -m comment --comment "kubernetes forwarding rules" -j KUBE-FORWARD
...
-A KUBE-FORWARD -m conntrack --ctstate INVALID -j DROP
```

Initial SYN packets go through to the Pod, SYN-ACK responses from local NodePort services are bpf_redirect'ed back out the physical interface to the client, but any follow-up packets from the client arriving at the node are considered invalid by netfilter's conntrack since the reply packet bypassed the stack, and thus dropped.

This commit takes care of adding `-i eni+` and `-o eni+` iptables rules to make sure world->container packets are never dropped in the stack.

```release-note
datapath: allow local NodePort traffic for `eni+` container interfaces with CNI chaining
```

### CODE REVIEW

**Review by pippolo84 — APPROVED** [review]
LGTM :100:

File: pkg/option/config.go [hunk 1]
```diff
@@ -1963,6 +1966,9 @@ type DaemonConfig struct {
 	// IPAM is the IPAM method to use
 	IPAM string
 
+	// Enable chaining with another CNI plugin.
+	CNIChainingMode string
```
Comment (joamaki) [inline pkg/option/config.go:1970]: should we register a flag for this and do `option.BindEnv` in `initializeFlags`?

**Review by joamaki — CHANGES_REQUESTED** [review]

**Review by borkmann — APPROVED** [review]
Modulo Jussi's comment, lgtm. 💯  I presume just the vp.GetString() won't search in env if not bound?

**Review by joamaki — APPROVED** [review]
LGTM :partying_face:

**Pull request merged**